### PR TITLE
[libtool] remove (last?) build system variable + add ltdl library

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  ~**libtool-ltdl/2.4.6**~==> **libtool/2.4.6**

~This library provides the `ltdl` library of libtool #659~

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by madebr** [comment]
@uilianries @danimtb I think the CI gave up and did not build all packages.

**Comment by danimtb** [comment]
@madebr why do you say that? I see the 12 logs there and it is the resulting packages from removing the compiler from package_id(), so it looks good to me 😕❓

**Comment by madebr** [comment]
@danimtb Oh! Thanks! That was a bug!

**Comment by danimtb** [comment]
That's better then 😆

File: recipes/libtool-ltdl/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+from contextlib import contextmanager
+
+
+class LibtoolLtDlConan(ConanFile):
+    name = "libtool-ltdl"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.gnu.org/software/libtool/"
+    description = "GNU libtool is a generic library support script."
+    topics = ("conan", "libtool", "configure", "library", "shared", "static")
+    license = ("GPL-2.0-or-later", "GPL-3.0-or-later")
```
Comment (uilianries) [inline recipes/libtool-ltdl/all/conanfile.py]:
```suggestion
    license = ("GPL-2.0-or-later")
```

Comment (uilianries) [reply recipes/libtool-ltdl/all/conanfile.py]: there is no reference about GPL-3.0 in the COPYING file, only GPL-2.0

File: recipes/libtool-ltdl/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return os.path.join(self.source_folder, "source_subfolder")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
```
Comment (uilianries) [inline recipes/libtool-ltdl/all/conanfile.py]: move to configure

**Comment by madebr** [comment]
@danimtb Please consider merging this pr. :relaxed:

**Comment by daravi** [comment]
I think this would be useful for pulseaudio.

**Comment by madebr** [comment]
Yes, they can be merged.
@uilianries @SSE4 @danimtb Can you review #1102 first?

**Comment by Croydon** [comment]
If we merge it into libtool than this PR can be closed I guess

**Comment by Croydon** [comment]
Please update the PR's title to avoid confusion, thanks 😄

**Comment by madebr** [comment]
>     * `libtool/2.4.6`: Generated 12 packages. [All logs here](https://c3i.jfrog.io/c3i/misc/logs/pr/807/11/libtool/2.4.6/)

@uilianries How come this low number of packages?

**Comment by madebr** [comment]
I pushed a new patch that removes some build system variables from the package.

Please review.
@danimtb  @uilianries  @SSE4

File: recipes/libtool/all/conanfile.py [hunk 3]
```diff
@@ -47,21 +61,26 @@ def _build_context(self):
...
 
+    @property
+    def _datarootdir(self):
+        return os.path.join(self.package_folder, "bin", "share")
+
+    def _my_unix_path(self, path):
+        if tools.os_info.is_windows:
+            return tools.unix_path(path)
+        else:
+            return path
```
Comment (danimtb) [inline recipes/libtool/all/conanfile.py:72]: This will be fixed in next Conan 1.25 release. unix_path() will have no effect on non-windows OSs

Comment (madebr) [reply recipes/libtool/all/conanfile.py:72]: What is the ETA of 1.25?

Comment (danimtb) [reply recipes/libtool/all/conanfile.py:72]: this week

Comment (madebr) [reply recipes/libtool/all/conanfile.py]: Removed this method.

**Comment by madebr** [comment]
@danimtb @uilianries @SSE4 @jgsogo 
Please review

**Pull request merged**